File: ignasr-app/src/main/java/limdongjin/ignasr/repository/MockIgniteRepository.java
package limdongjin.ignasr.repository;

import reactor.core.publisher.Mono;

import java.util.concurrent.ConcurrentHashMap;

public class MockIgniteRepository implements IgniteRepository{
    public ConcurrentHashMap<String, ConcurrentHashMap<Object, Object>> name2cache;

    public MockIgniteRepository() {
        this.name2cache = new ConcurrentHashMap<>();
    }

    @Override
    public <K, V> Mono<Mono<K>> putAsync(String cacheName, K key, V value) {
        Mono<K> kMono = Mono.just(key)
            .doOnNext(k -> {
                ConcurrentHashMap<K, V> cache;
                if(name2cache.containsKey(cacheName)){
                    cache = (ConcurrentHashMap<K, V>) name2cache.get(cacheName);
                }else {
                    cache = new ConcurrentHashMap<K, V>();
                    name2cache.put(cacheName, (ConcurrentHashMap<Object, Object>) cache);
                }
                cache.put(key, value);
            })
        ;
        Mono<Mono<K>> monoMonoK = Mono.just(kMono);
        return monoMonoK;
    }

    @Override
    public <K, V> K put(String cacheName, K key, V value) {
        ConcurrentHashMap<K, V> cache;
        if(name2cache.containsKey(cacheName)){
            cache = (ConcurrentHashMap<K, V>) name2cache.get(cacheName);
        }else {
            cache = new ConcurrentHashMap<K, V>();
            name2cache.put(cacheName, (ConcurrentHashMap<Object, Object>) cache);
        }
        cache.put(key, value);
        return key;
    }

    @Override
    public <K, V> V get(String cacheName, K key) {
        System.out.println(name2cache);
        ConcurrentHashMap<K, V> cache;
        if(name2cache.containsKey(cacheName)){
            cache = (ConcurrentHashMap<K, V>) name2cache.get(cacheName);
        }else {
            cache = new ConcurrentHashMap<K, V>();
            name2cache.put(cacheName, (ConcurrentHashMap<Object, Object>) cache);
        }
        return cache.get(key);
    }

    @Override
    public <K, V> int size(String cacheName) {
        int size;
        ConcurrentHashMap<K, V> cache;
        if(name2cache.containsKey(cacheName)){
            cache = (ConcurrentHashMap<K, V>) name2cache.get(cacheName);
        }else {
            cache = new ConcurrentHashMap<K, V>();
            name2cache.put(cacheName, (ConcurrentHashMap<Object, Object>) cache);
        }
        size = cache.size();
        return size;
    }
}

File: ignasr-app-kt/src/main/kotlin/limdongjin/ignasr/repository/MockIgniteRepository.kt
package limdongjin.ignasr.repository

import reactor.core.publisher.Mono
import java.util.concurrent.ConcurrentHashMap


class MockIgniteRepository: IgniteRepository {
    val name2cache: ConcurrentHashMap<String, ConcurrentHashMap<Any, Any>> =
        ConcurrentHashMap();

    override fun <K : Any, V : Any> putAsync(cacheName: String, key: K, value: V): Mono<Mono<K>?> {
        val kMono: Mono<K> =
            Mono.just<K>(key)
                .doOnNext { k: K ->
                    val cache = name2cache.getOrPut(cacheName) { ConcurrentHashMap() }
                    cache!![key] = value
                }
        return Mono.just<Mono<K>>(kMono)
    }

    override fun <K: Any, V: Any> put(cacheName: String, key: K, value: V): K {
        val cache = name2cache.getOrPut(cacheName) { ConcurrentHashMap() }
        cache[key] = value
        return key
    }

    override fun <K: Any, V: Any> get(cacheName: String, key: K): V {
        val cache = name2cache.getOrPut(cacheName) { ConcurrentHashMap() }
        return cache[key]!! as V
    }

    override fun <K: Any, V: Any> size(cacheName: String): Int {
        val cache = name2cache.getOrPut(cacheName) { ConcurrentHashMap() }
        return cache.size
    }
}